src/common/context/forms/index.tsx

```diff
@@ -64,25 +65,34 @@ export const FormsContextProvider: FunctionComponent = ({ children }) => {
     setActiveFormIndex(newIndex);
   };
 
-  const newPopulatedForm = (templateIndex: number, data: Array<FormEntry>): void => {
-    const newFormTemplate = config?.forms[templateIndex];
-    const newFormName = newFormTemplate?.name ?? "Document";
-    const formEntries: FormEntry[] = [];
-    const extension = config?.forms[templateIndex]?.extension ?? "tt";
-    for (let index = 0; index < data.length; index++) {
-      formEntries.push({
-        templateIndex,
-        data: {
-          formData: data[index],
-          schema: newFormTemplate?.schema,
-        },
-        fileName: data[index].fileName ?? `${newFormName}-${forms.length + 1 + index}`,
-        ownership: data[index].ownership ?? { beneficiaryAddress: "", holderAddress: "" },
-        extension: extension,
-      });
+  const newPopulatedForm = (templateIndex: number, data: Array<FormEntry>, fileName?: string): void => {
+    try {
+      const newFormTemplate = config?.forms[templateIndex];
+      const newFormName = newFormTemplate?.name ?? "Document";
+      const formEntries: FormEntry[] = [];
+      const extension = config?.forms[templateIndex]?.extension ?? "tt";
+      const filenameTemplate = fileName ? _.template(fileName) : undefined;
+
+      for (let index = 0; index < data.length; index++) {
+        formEntries.push({
+          templateIndex,
+          data: {
+            formData: data[index],
+            schema: newFormTemplate?.schema,
+          },
+          fileName: filenameTemplate ? filenameTemplate(data[index]) : `${newFormName}-${forms.length + 1 + index}`,
+          ownership: data[index].ownership ?? { beneficiaryAddress: "", holderAddress: "" },
+          extension: extension,
+        });
+      }
+      setForms([...forms, ...formEntries]);
+      setActiveFormIndex(forms.length);
+    } catch (e) {
+      if (e instanceof ReferenceError) {
+        throw new Error("filename in configuration file could not be found in data");
```

Comment:
maybe can change it to: 
"failed to interpolate data properties, could not find data properties in configuration file"

u see if its better? if not need think again lol

Reply:
done.